### PR TITLE
Add HTML Popup Class to element.py

### PR DESCRIPTION
**Pull request by samchorlton**
Add HTML Popup functionality to element.py class.

### CODE REVIEW

**Comment by BibMartin** [comment]
Hi @samchorlton,
Thanks for your contribution ; you're the first one on branca !

It seems what you are proposing is like an improvement of `branca.element.Html` (or `branca.element.Div` ; in fact they are quite similar). Do you think it would be possible to change one of them instead of creating a third similar object ?

Maybe one shall add an option in `Html` to replace `{{this.data|e}}` by `{{this.data}}` ; something like:

```python
{% if this.escape %}{{this.data}}{% else %}{{this.data|e}}{% endif %}
```

And then add your (awesome) trick to make the ratio customizable. (btw: if it were possible to handle the `if` statement in the Template, is would avoid reprogramming the `render` method).

**Comment by samchorlton** [comment]
I have added a flag to the existing html class which means if the text is enclosed in <html_code></html_code> it will be rendered as code rather than printed to string.


How does that look?

**Comment by BibMartin** [comment]
That's good. But why not simply a keyword argument `html=True` ?

PS: the checks are failing because of pep8 verification ; you can run 

    find . -type f -name "*.py" ! -name 'conf.py' | xargs flake8 --max-line-length=100

to find out what's wrong (it requires `falke8`).

**Comment by samchorlton** [comment]
I decided to set it up to be enclosed with html tags so that you could utilise the existing simple_marker function used in folium. 

Happy to use your suggestion would just require altering simple_marker as well. What do you suggest?

Will check pep8 verification running on windows at the moment..

**Comment by BibMartin** [comment]
> Happy to use your suggestion would just require altering simple_marker as well. What do you suggest?

Yes, I would prefer to add it eplicitely in both methods (and document them accordingly) instead of using the *trick* of `<html_code>`. It will be easier for other users to understand it.

> Will check pep8 verification running on windows at the moment..

Maybe not, provided this line is a bash line. You would need to *translate* it into a windows one or install a bash interpreter. At least you can run `flake8 somefile.py` on all files you have changed, and verify that no line has more than 100 chars.

**Comment by samchorlton** [comment]
Have made the changes as suggested and will now commit additional changes to folium.

**Comment by BibMartin** [comment]
Great!
I'll be anoying, but:

* `script` needs to be a keyword argument with default value `False` to ensure backward-compatibility
* The docstring of the class needs to be updated

**Comment by samchorlton** [comment]
Do I need to set the keyword argument value in the folium or the branca code?

**Comment by BibMartin** [comment]
> Do I need to set the keyword argument value in the folium or the branca code?

In both. In folium, you'll set `script=False` in the method definition, and then call `Html(...script=script)` in the call to branca.

**Comment by samchorlton** [comment]
That should be the changes you suggested made.

File: branca/element.py [hunk 1]
```diff
@@ -428,6 +430,8 @@ class Html(Element):
     ----------
     data : str
         The HTML data to be embedded.
+    script : bool
+        The datatype of the HTML data
```
Comment (BibMartin) [inline branca/element.py]: It shall be more explicit ; something like:
```If True, data will be embedded without escaping (suitable for embedding html-ready code)```

**Comment by BibMartin** [comment]
That's ok for me. You need to replace tabs by spaces to fix pep8.

Then can you squash your commits into a signle one, and I'll merge.

**Comment by samchorlton** [comment]
Hopefully that will fix it.

**Comment by BibMartin** [comment]
Here's how to squash commits:
http://www.git-scm.com/book/en/v2/Git-Tools-Rewriting-History

**Comment by samchorlton** [comment]
I have tried to do a squash commit but not done it before so let me know if that's right or I'll try again.

**Comment by BibMartin** [comment]
That's almost it. You've successfully rebased (all your modifications are summarized in 90f6b13).
But this commit has been added to others instead of replacing them.

To correct this, you can do:

    git checkout 90f6b13
    git checkout -b new_branch
    git push -f your_remote new_branch:master

where `your_remote` is the remote name of your repository (`origin` if you cloned your fork). Note the `-f` option that's forces the `push` to overwrite previous changes (this is not a good practice on shared repositories, but you're on your fork) ; I think this may be what was missing in your rebase operation.

btw: don't worry about doing some wrong manipulation, I've cloned you branch as a backup.

**Comment by samchorlton** [comment]
So just tried what you said thank you for giving me a hand not done a whole lot of git in the past. Let me know if that worked or not. I wasn't sure what directory level I needed to be at the time.

**Comment by BibMartin** [comment]
That's perfect ; I merge it immediately.

Thanks for the contribution, and welcome in the family :tada:

**Pull request merged**